### DISCUSSION TITLE
CSRF - Implementation options?

### DISCUSSION THREAD

**Discussion by rohitkumbhar** [micronaut-projects/micronaut-core#7951] [General]
I've been trying to build simple CSRF protection using cookie + param value comparison as shown in [security docs](https://micronaut-projects.github.io/micronaut-security/latest/guide/#authorization-code). I can't make it a generic implementation i.e. `_csrf` value from url param or body because HttpFilter do not read the body: [Related Issue](https://github.com/micronaut-projects/micronaut-core/issues/1113)

Does anyone have any workarounds for this?

**Comment by yawkat** [comment]
there's currently no way to get the body in a filter, but we are looking at this. you could still implement this with a netty channel handler but it's probably more effort than it's worth.
